packages/core/contracts/financial-templates/expiring-multiparty/PricelessPositionManager.sol

```diff
@@ -604,6 +609,18 @@ contract PricelessPositionManager is FeePayer, AdministrateeInterface {
         return;
     }
 
+    function trimExcess(IERC20 token) external fees() nonReentrant() {
+        FixedPoint.Unsigned memory balance = FixedPoint.Unsigned(collateralCurrency.balanceOf(address(this)));
+        if (address(token) == address(collateralCurrency)) {
+            // If it is the collateral currency, send only the amount that the contract is not tracking.
+            // Note: this could be due to rounding error or balance-chainging tokens, like aTokens.
+            token.safeTransfer(beneficiary, balance.sub(pfc()).rawValue);
```

Comment:
This makes sense to me. PFC is simply the `rawTotalPositionCollateral` adjusted for fees. This is updated every time someone deposits or withdraws from the contract. If the balance shifts through some other action (an a-token or someone depositing collateral directly to the emp) then the balance of call will be higher than the `pfc` and this function will trip any of that excess balance off.

In theory, this could also let us recover funds that are incorrectly sent to the contract.

Reply:
Here you do not perform any checks on who `beneficiary` is assigned to. It'll default to the 0x000 address as a result. Because this function has no access control someone could grief the EMP if it has any excess collateral by sending it to the 0x000 address. I'd consider either adding a check that enforces that the `beneficiary` is not the 0 address or perhaps setting it to the `Store`?

Reply:
Done. I added it to the validation done in the creator so the check wouldn't affect the bytecode size.